mailet/standard/src/main/java/org/apache/james/transport/mailets/StripAttachment.java

```diff
@@ -63,156 +68,158 @@
+        decodeFilename = getBooleanParameter(getInitParameter(DECODE_FILENAME_PARAMETER_NAME), DECODE_FILENAME_DEFAULT_VALUE);
+        String replaceFilenamePattern = getInitParameter(REPLACE_FILENAME_PATTERN_PARAMETER_NAME);
+        if (replaceFilenamePattern != null) {
+            filenameReplacingPatterns = new PatternExtractor().getPatternsFromString(replaceFilenamePattern);
+        } else {
+            filenameReplacingPatterns = ImmutableList.of();
         }
 
-        String toLog = String.format("StripAttachment is initialised with regex pattern [%s / %s]",
-                patternString, notpatternString);
-        if (directoryName != null) {
-            toLog += String.format(" and will save to directory [%s]", directoryName);
+        logConfiguration();
+    }
+
+    private Pattern regExFromParameter(String patternParameterName) throws MailetException {
+        String patternString = getInitParameter(patternParameterName);
+        try {
+            if (patternString != null) {
+                return Pattern.compile(patternString);
+            }
+            return null;
+        } catch (Exception e) {
+            throw new MailetException("Could not compile regex [" + patternString + "].");
         }
-        if (attributeName != null) {
-            toLog += String.format(" and will store attachments to attribute [%s]", attributeName);
+    }
+
+    private void createDirectory() throws MailetException {
+        try {
+            FileUtils.forceMkdir(new File(directoryName));
+        } catch (Exception e) {
+            throw new MailetException("Could not create directory [" + directoryName + "].", e);
         }
-        log(toLog);
     }
 
+      private void logConfiguration() {
+      StringBuilder logMessage = new StringBuilder();
+      logMessage.append("StripAttachment is initialised with regex pattern [");
+      if (regExPattern != null) {
+          logMessage.append(regExPattern.pattern());
+      } else {
+          logMessage.append("");
+      }
+      logMessage.append(" / ");
+      if (notregExPattern != null) {
+          logMessage.append(notregExPattern.pattern());
+      } else {
+          logMessage.append("");
+      }
+      logMessage.append("]");
+
+      if (directoryName != null) {
+          logMessage.append(" and will save to directory [");
+          logMessage.append(directoryName);
+          logMessage.append("]");
+      }
+      if (attributeName != null) {
+          logMessage.append(" and will store attachments to attribute [");
+          logMessage.append(attributeName);
+          logMessage.append("]");
+      }
+      log(logMessage.toString());
+    }
+    
     /**
-     * Service the mail: scan it for attchemnts matching the pattern, store the
-     * content of a matchin attachment in the given directory.
+     * Service the mail: scan it for attachments matching the pattern, store the
+     * content of a matching attachment in the given directory.
```

Comment:
As told by benoit, that's not the only things the mailet is doing, so add a comment about the attribute things.